tests/failpoints_cases/test_storage.rs

```diff
@@ -39,32 +38,45 @@ fn test_storage_1gc() {
     let mut storage = Storage::from_engine(engine.clone(), &config, read_pool).unwrap();
     storage.start(&config).unwrap();
     fail::cfg(snapshot_fp, "pause").unwrap();
-    fail::cfg(batch_snapshot_fp, "pause").unwrap();
     let (tx1, rx1) = channel();
+    // Schedule `GC_MAX_PENDING` GC requests.
+    for _i in 0..GC_MAX_PENDING {
+        let tx1 = tx1.clone();
+        storage
+            .async_gc(ctx.clone(), 1, box move |res: storage::Result<()>| {
+                assert!(res.is_ok());
+                tx1.send(1).unwrap();
+            })
+            .unwrap();
+    }
+    // Sleep to make sure the failpoint is triggered.
+    thread::sleep(Duration::from_millis(2000));
+    // Schedule one more request. So that there is a request being processed and
+    // `GC_MAX_PENDING` requests in queue.
     storage
         .async_gc(ctx.clone(), 1, box move |res: storage::Result<()>| {
             assert!(res.is_ok());
             tx1.send(1).unwrap();
         })
         .unwrap();
-    // Sleep to make sure the failpoint is triggered.
-    thread::sleep(Duration::from_millis(2000));
-    // Old GC command is blocked at snapshot stage, the other one will get ServerIsBusy error.
+
+    // Old GC commands are blocked, the new one will get GCWorkerTooBusy error.
```

Comment:
Does new gc can block at getting snapshot?

Reply:
The first GC task is blocked at getting snapshot, and the others are blocked in the `Worker`